schema/object.go

```diff
@@ -163,12 +163,9 @@ func (o *ObjectSchema) serializeMap(data map[string]any) (any, error) {
 		if err != nil {
 			return nil, ConstraintErrorAddPathSegment(err, k)
 		}
-		defaultValue, hasDefaultValue := o.defaultValues[k]
-		if !hasDefaultValue && defaultValue != serializedValue {
-			rawData[k] = serializedValue
-		}
+		rawSerializedData[k] = serializedValue
```

Comment:
Not to be impertinent or anything, but how come you can make this change without modifying or removing any tests?  (That suggests that there are tests missing...which should probably be added by this PR...no?  🙂)

Reply:
That is correct. The defaults behavior is not well defined in that regard. So yeah, I agree; tests are needed. I'll do that tomorrow.

Reply:
That is correct. The defaults behavior is not well defined in that regard. So yeah, I agree; tests are needed. I'll do that tomorrow.

Reply:
Tests added. And I found out from those tests that I needed to make the change to one other spot.